### DISCUSSION TITLE
Where does the ZWS encoding actually happen?

### DISCUSSION THREAD

**Discussion by SohamG** [zws-im/zws#567] [Q&A] [answered]
Hello,

I was looking through the source code trying to understand how the url is encoded in the unicode zero width space character aka U+200B.

I got to this file (https://github.com/zws-im/zws/blob/main/apps/api/src/urls/urls.service.ts) but that file doesn't seem to mention or reference said unicode character? Please can someone help me understand this?

I apologize in advance if posting this here isn't the most convenient.

Thanks

**Comment by jonahsnider** [comment, marked as answer]
The zero-width shortened string is generated in [this method](https://github.com/zws-im/zws/blob/24af901171362b51c364b556e28318849a49a369/apps/api/src/urls/urls.service.ts#L144). It uses characters from the config defined in [this file](https://github.com/zws-im/zws/blob/24af901171362b51c364b556e28318849a49a369/apps/api/src/urls/urls-config.service.ts).

**Reply by SohamG** [reply]
Thanks for the reply! I was under the assumption that U+200B was the *only* zero width space in unicode. Turns out that is not the case.
